File: src/Dto/Project.Dto.Http/Score/ScoreDto.cs
namespace Project.Dto.Http.Score;

public class ScoreDto
{
    public ScoreDto(Guid id, Guid employeeId, Guid authorId, Guid positionId, DateTimeOffset createdAt,
        int efficiencyScore,
        int engagementScore, int competencyScore)
    {
        Id = id;
        EmployeeId = employeeId;
        AuthorId = authorId;
        PositionId = positionId;
        CreatedAt = createdAt;
        EfficiencyScore = efficiencyScore;
        EngagementScore = engagementScore;
        CompetencyScore = competencyScore;
    }

    public Guid Id { get; set; }
    public Guid EmployeeId { get; set; }
    public Guid AuthorId { get; set; }
    public Guid PositionId { get; set; }
    public DateTimeOffset CreatedAt { get; set; }
    public int EfficiencyScore { get; set; }
    public int EngagementScore { get; set; }
    public int CompetencyScore { get; set; }
}

File: front/src/app/dto/score.dto.ts
/**
 * DTO for score received from API
 */
export interface ScoreDto {
    id: string;
    employeeId: string;
    authorId: string;
    positionId: string;
    createdAt: string; // ISO date-time string (DateTimeOffset from backend)
    efficiencyScore: number; // 1-5
    engagementScore: number; // 1-5
    competencyScore: number; // 1-5
  }
  
  /**
   * DTO for creating a score
   */
  export interface CreateScoreDto {
    employeeId: string;
    authorId: string;
    positionId: string;
    createdAt: string; // ISO date-time string (DateTimeOffset from backend)
    efficiencyScore: number; // 1-5
    engagementScore: number; // 1-5
    competencyScore: number; // 1-5
  }
  
  /**
   * DTO for updating a score
   */
  export interface UpdateScoreDto {
    createdAt?: string; // ISO date-time string (DateTimeOffset from backend)
    efficiencyScore?: number; // 1-5
    engagementScore?: number; // 1-5
    competencyScore?: number; // 1-5
  }
  